hnswlib/space_l2.h

```diff
@@ -19,7 +19,38 @@ namespace hnswlib {
         return (res);
     }
 
-#if defined(USE_AVX)
+#if defined(USE_AVX512)
+
+    // Favor using AVX512 if available.
+    static float
+    L2SqrSIMD16Ext(float *pVect1, float *pVect2, size_t qty) {
```

Comment:
Sorry for a long response. It seems `float` type here obstructs the compilation.
It should be `void *`

Reply:
> Sorry for a long response. It seems `float` type here obstructs the compilation. It should be `void *`

Thanks for catching this. I have fixed it.

Reply:
I've tried and there other errors with `constant  void *` to `void *` conversion (I guess it should be made const void * down the stack).
Can you please fix it and test it with `pip install .` and `python -m unittest discover --start-directory python_bindings/tests --pattern "*_test*.py"` (to run the tests) on a linux system.

Reply:
Thanks for the help! I fixed those errors. I was able to compile it and test it without errors following your instructions.